examples/react/trimmed/src/components/views/nozzle-paa.tsx

```diff
@@ -0,0 +1,563 @@
+      let highlightCol;
+
+      // Robust check: Ensure predicate exists AND is not an empty array
+      const hasPred =
+        highlightPred &&
+        (!Array.isArray(highlightPred) || highlightPred.length > 0);
+
+      if (hasPred) {
+        // Ensure the predicate is a valid SQL Node for interpolation.
+        // If highlightPredicate is an array (implicit AND), wrap it.
+        const safePredicate = Array.isArray(highlightPred)
+          ? mSql.and(...highlightPred)
+          : highlightPred;
+
+        highlightCol = mSql.max(
+          mSql.sql`CASE WHEN ${safePredicate} THEN 1 ELSE 0 END`,
+        );
+      } else {
+        highlightCol = mSql.literal(1);
+      }
+
+      const q = mSql.Query.from(TABLE_NAME)
+        .select({
+          [safeId]: groupKey,
+          metric: metric === '*' ? aggFn() : aggFn(metric),
+          // Export computed highlight column
+          __is_highlighted: highlightCol,
+        })
+        .groupby(groupKey);
+
+      if (filter) {
+        q.where(filter);
+      }
+      if (where) {
+        q.where(where);
+      }
+
+      return q;
+    },
+    [groupBy, metric, aggFn, where, safeId],
+  );
+
+  const baseTableOptions = useMemo(
+    () => ({
+      initialState: {
+        sorting: [{ id: 'metric', desc: true }],
+        pagination: { pageSize: 10 },
+      },
+    }),
+    [],
+  );
+
+  const mosaicOptions = useMemo(
+    () => ({
+      table: queryFactory,
+      // FILTER BY Inputs AND Detail Table
+      filterBy: $summaryContext,
+      // HIGHLIGHT BY Peers (Cross-filtering)
+      highlightBy: $crossFilter,
+      // NEW: Tell Core we handled it manually
+      manualHighlight: true,
+      columns: [],
+      tableOptions: baseTableOptions,
+    }),
+    [queryFactory, baseTableOptions],
+  );
+
+  const { tableOptions, client } = useMosaicReactTable(mosaicOptions);
+  const selectedValue = useSelectionValue($crossFilter, client);
+
+  const columns = useMemo(
+    () => [
+      {
+        id: 'select',
+        header: '',
+        size: 30,
+        enableSorting: false,
+        enableColumnFilter: false,
+        enableHiding: false,
+        cell: ({ row }: any) => {
+          const rowVal = row.getValue(groupBy);
+          const isChecked = selectedValue === rowVal;
+          return (
+            <input
+              type="checkbox"
+              checked={isChecked}
+              readOnly
+              className="cursor-pointer size-4"
+            />
+          );
+        },
+      },
+      {
+        id: groupBy,
+        accessorKey: safeId,
+        header: title,
+        enableColumnFilter: false,
+      },
+      {
+        id: 'metric',
+        accessorKey: 'metric',
+        header: metricLabel,
+        cell: (info: any) => info.getValue()?.toLocaleString(),
+        enableColumnFilter: false,
+      },
+    ],
+    [groupBy, title, metricLabel, safeId, selectedValue],
+  );
+
+  useEffect(() => {
+    client.updateOptions({
```

Comment:
Why is updateOptions being called at all here? 

The current logic you've written here is.

1. Memoize the Mosaic Data Table Options.
2. Create Mosaic Data Table Options using the hook.
3. Memoize the Column definitions.
4. Update the Mosaic Data Table Client with the Columns using `updateOptions`.

This makes no sense to me. This should all, just be happening above in `mosaicOptions`.